azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java

```diff
@@ -1129,6 +1132,32 @@ private void createPod(final int executionId)
     if (flowParam != null && !flowParam.isEmpty()) {
       logger.info("ExecId: {}, Flow Parameters are: {}", executionId, flowParam);
     }
+
+    // Check if flow params has user.to.proxy If it does, skip the next step since this overrides
+    // all the project and node properties.
+    HashSet<String> proxyUsersMap = new HashSet<>();
+    if (flowParam != null && flowParam.containsKey("user.to.proxy")) {
+      proxyUsersMap.add(flowParam.get("user.to.proxy"));
+    }
```

Comment:
Overwriting a USER_TO_PROXY value at runtime doesn't mean that's the value that is going to be used in the end, there are other ways this value can change which have higher precedence than runtime properties.

Reply:
ack. removing the if condition and adding it to the list of proxy users. thanks for catching this.